pippo-core/src/test/java/ro/pippo/core/RouteGroupTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ro.pippo.core;
+
+import org.junit.Test;
+import ro.pippo.core.route.Route;
+import ro.pippo.core.route.RouteGroup;
+import ro.pippo.core.route.RouteMatch;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author ScienJus
+ * @date 16/2/9.
+ */
+public class RouteGroupTest {
```

Comment:
If we add `addRouteGroup` in `Router` we don't need this new class and we can add these new tests in DefaultRouterTest. Please use the approach from DefaultRouterTest (you have a router instance) and remove the code with `new Application() { ... }`